Authentication/CloudPakForData/CloudPakForDataAuthenticator.cs

```diff
@@ -131,6 +134,7 @@ private void GetToken()
 
         private void OnGetToken(DetailedResponse<CloudPakForDataTokenResponse> response, IBMError error)
         {
+            isDoneInitializing = true;
```

Comment:
Should `isDoneInitializing` be called after we throw the error?

Reply:
Well, yes.

If we move the new line a few lines down (inside `else` statement) so it's not called when an error happens, then there would be no way left for a consumer class to know if an error happened.

If you think this is not clear enough, maybe we should have an `IsWaiting` field instead, it would have the opposite value to `IsDoneInitializing`, and maybe makes thing more clear?